### PR TITLE
enh(Right Panel): Adapt right panel

### PR DESCRIPTION
**Pull request by Thebarda**
This PR renames "Right Panel" into "Listing Panel" (that has a fixed width) and creates a new "Listing Panel" (that fits in hiw own parents). That last one should be used within "Listing Page"

Listing Panel
storyshots-test-js-image-story-shots-listing-panel-normal-1-snap

Page Panel

storyshots-test-js-image-story-shots-page-panel-normal-1-snap

### CODE REVIEW

**Comment by bdauria** [comment]
@Thebarda I think it would make more sense to name the two panels the other way around (SlidePanel -> ListingPanel, as it's local to the Listing, RightPanel -> PagePanel, as it's global to the page)

File: src/index.ts [hunk 1]
```diff
@@ -100,7 +100,8 @@ export { default as SubmenuItems } from './Submenu/SubmenuHeader/SubmenuItems';
 export { default as SubmenuItem } from './Submenu/SubmenuHeader/SubmenuItem';
 
 export { default as MultiSelectEntries } from './MultiSelectEntries';
-export { default as RightPanel } from './RightPanel';
+export { default as PagePanel } from './ListingPanel';
+export { default as ListingPanel } from './PagePanel';
```
Comment (bdauria) [inline src/index.ts:104]: Directories should be renamed according to the Component names

File: src/ListingPanel/index.jsx [hunk 2]
```diff
@@ -0,0 +1,164 @@
+import React, { useState, useEffect } from 'react';
```
Comment (kduret) [inline src/ListingPanel/index.jsx:1]:
```suggestion
import * as React from 'react';
```

File: src/ListingPanel/index.jsx [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+const PagePanel = ({
+  active,
+  Header,
+  secondaryPanelComponent,
+  onSecondaryPanelClose,
+  Sections,
+  onClose,
+  loading,
+}) => {
+  const [secondaryPanelActive, setSecondaryPanelActive] = useState(false);
```
Comment (kduret) [inline src/ListingPanel/index.jsx:59]:
```suggestion
  const [secondaryPanelActive, setSecondaryPanelActive] = React.useState(false);
```

File: src/ListingPanel/index.jsx [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+  const classes = useStyles({
+    active: secondaryPanelActive,
+  });
+
+  useEffect(() => {
```
Comment (kduret) [inline src/ListingPanel/index.jsx:64]:
```suggestion
  React.useEffect(() => {
```

File: src/PagePanel/index.test.jsx [hunk 5]
```diff
@@ -0,0 +1,56 @@
+import React from 'react';
```
Comment (kduret) [inline src/PagePanel/index.test.jsx:1]:
```suggestion
import * as React from 'react';
```

**Comment by bdauria** [comment]
> image
> is this space normal ? Is it the scrollbar ?

It's actually used to close the secondary panel when open. But it's not very intuitive though, as it's always displayed

File: src/ListingPanel/index.jsx [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+  overflowY: 'hidden',
+});
+
+const MainPanel = styled(Box)({
+  width: 540,
+  overflowY: 'auto',
+});
+
+export const SecondaryPanelBar = styled(Box)({
+  border: '1px solid #D6D6D8',
```
Comment (kduret) [inline src/ListingPanel/index.jsx:28]: could you use material-ui grey palette, like it's done in wizard ?
* https://github.com/centreon/centreon-ui/blob/master/src/Wizard/index.jsx#L46
* https://github.com/centreon/centreon-ui/blob/master/src/Wizard/ActionBar.jsx#L14

File: src/ListingPanel/index.jsx [hunk 7]
```diff
@@ -0,0 +1,164 @@
...
+  width: 20,
+  cursor: 'pointer',
+});
+
+const useStyles = makeStyles((theme) => ({
+  secondaryPanel: {
+    width: ({ active }) => (active ? 500 : 0),
+    transition: '.1s ease-in-out',
+    overflow: 'hidden',
+    backgroundColor: '#c7c8c9',
```
Comment (kduret) [inline src/ListingPanel/index.jsx:38]: same comment as above

File: src/PagePanel/index.jsx [hunk 8]
```diff
@@ -54,7 +54,7 @@ const MainPanel = styled(Box)({
   marginBottom: topHeight,
 });
 
-const SecondaryPanelBar = styled(Box)({
+export const SecondaryPanelBar = styled(Box)({
   border: '1px solid #D6D6D8',
```
Comment (kduret) [inline src/PagePanel/index.jsx:58]: same comment as above (even if you did not modify this line)

File: src/ListingPanel/index.jsx [hunk 9]
```diff
@@ -0,0 +1,164 @@
...
+    <SlidePanel
+      header={header}
+      content={
+        active && (
+          <Body display="flex" flexDirection="row" flexGrow={1}>
+            {loading ? (
+              <div className={classes.loading}>
+                <CircularProgress />
+              </div>
+            ) : (
```
Comment (bdauria) [inline src/ListingPanel/index.jsx:108]: We can use the ContentWithCircularProgress Component here.

File: src/PagePanel/index.test.jsx [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+    expect(getByText('Expand me')).toBeInTheDocument();
+    expect(getByText('Expandable Section')).toBeInTheDocument();
+  });
+
+  it('displays secondary Panel when secondary Panel bar is clicked', () => {
+    const secondaryPanelComponent = <>Secondary Panel</>;
+
+    const { baseElement, getByText, queryByText } = render(
+      <ListingPanel
+        Header={<>Header</>}
```
Comment (kduret) [inline src/PagePanel/index.test.jsx:38]:
```suggestion
        Header={Header}
```

**Pull request merged**